core/common/src/main/java/alluxio/util/io/FileUtils.java

```diff
@@ -134,6 +135,51 @@ public static short translatePosixPermissionToMode(Set<PosixFilePermission> perm
     return (short) mode;
   }
 
+  /**
+   * Translate mode to posix file permissions.
+   *
+   * @param mode the file mode
+   * @return posix file permissions
+   */
+  public static Set<PosixFilePermission> translateModeToPosixPermissions(int mode)
+      throws IOException {
+    Set<PosixFilePermission> perms = new HashSet<>();
+    // add owners permission
+    if (mode < 0) {
+      throw new IOException("Mode can not be a negative value");
+    }
```

Comment:
This should not throw IOE:
```suggestion
    Preconditions.checkArgument(mode >=0, "Mode can not be a negative value");
```